sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -500,16 +500,18 @@ def get_tx_power(self):
         '''
         This function returns TX output power in mW on each media lane
         '''
-        tx_power_support = self.get_tx_power_support()
-        if tx_power_support is None:
-            return None
-
         tx_power = ["N/A" for _ in range(self.NUM_CHANNELS)]
 
+        tx_power_support = self.get_tx_power_support()
+        if not tx_power_support:
+            return tx_power
+
         if tx_power_support:
             tx_power = self.xcvr_eeprom.read(consts.TX_POWER_FIELD)
             if tx_power is not None:
                 tx_power =  [tx_power['OpticalPowerTx%dField' %i] for i in range(1, self.NUM_CHANNELS+1)]
+            else:
+                return None
```

Comment:
this else is not required? line 512 `tx_power = None` already

Reply:
Updated